src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddleware.cs

```diff
@@ -113,11 +113,21 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
                 };
                 context.Features.Set<IExceptionHandlerFeature>(exceptionHandlerFeature);
                 context.Features.Set<IExceptionHandlerPathFeature>(exceptionHandlerFeature);
-                context.Response.StatusCode = 500;
+                context.Response.StatusCode = StatusCodes.Status500InternalServerError;
                 context.Response.OnStarting(_clearCacheHeadersDelegate, context.Response);
 
                 await _options.ExceptionHandler(context);
 
+                if (context.Response.StatusCode == StatusCodes.Status404NotFound)
+                {
+                    // Reset status and clear headers
```

Comment:
This might not be necessary. For example, Kestrel would set the 500 status by default when an exception is encountered. But theoretically it wouldn't hurt to set it here right?